### ISSUE TITLE
False `print_literal` on `println!` with named arguments

### ISSUE THREAD

**Issue by commandline** [rust-lang-nursery/rust-clippy#2956]
I started receiving a false positive on the `print_literal` lint with `6a1c0637c 2018-07-23` nightly.

I use the named arguments to `println!` that it supports because it uses `format!`. Now I get the `print_literal` lint all over my code base for code like this:

```
        println!(
            "
Your {prefix} pipeline has been created in {output}.

You can run

jsonnet --jpath {mstk}/jsonnet/ {output}

to test that the file evaluates correctly.
",
            prefix = prefix,
            mstk = shared_dev.mstk_home,
            output = generator.output_path(),
        );
```
Where `prefix` is one of my enums that implements `std::fmt::Display`, `shared_dev.mstk_home` is a `String` field on one of my own structs, and the return of the `generator.output_path()` function is a `String`.

If I change these to use indices, the lint clears however changing all of these is a lot of makework for what seems like a bug and erodes readability.

**Comment by oli-obk** [comment]
Oh. haha, "copy paste and forget"-failure:

https://github.com/rust-lang-nursery/rust-clippy/blob/master/clippy_lints/src/write.rs#L292

needs a check for the `rhs` to see whether it's an `ExprKind::Lit(_)`

**Comment by commandline** [comment]
I think I understand what you are saying needs changing. I'd be happy to try to tackle this with a little help.

**Comment by oli-obk** [comment]
Basically wrap https://github.com/rust-lang-nursery/rust-clippy/blob/master/clippy_lints/src/write.rs#L293-L310 in another `if let ExprKind::Lit(_) = rhs.node` and add some tests.

If you have any questions don't hesitate to ask here, on irc on discord in the #clippy channels.

**Comment by commandline** [comment]
Awesome, thanks!

**Comment by commandline** [comment]
Are there existing test to add to?

**Comment by flip1995** [comment]
There is https://github.com/rust-lang-nursery/rust-clippy/blob/master/tests/ui/print_literal.rs. You can just add the tests here!

**Comment by commandline** [comment]
Opened #2963, let me know if the changes to tests make sense. The fix is literally what @oli-obk described which makes sense to me.

I added to the write_literal.rs as well as it seems like that exercises the same code.

**Comment by oli-obk** [comment]
fixed in #2963
